src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -13,6 +13,32 @@ public static partial class Path
 
         public static char[] GetInvalidPathChars() => new char[] { '\0' };
 
+        // Checks if the given path is available for use.
+        public static bool Exists([NotNullWhen(true)] string? path)
+        {
+            try
+            {
+                if (path == null)
+                {
+                    return false;
+                }
+
+                if (path.Length == 0)
+                {
+                    return false;
+                }
+
+                ReadOnlySpan<char> fullPath = GetFullPath(path).AsSpan();
```

Comment:
do we need to trim any ending separator as current impl does?
https://github.com/dotnet/runtime/blob/6dfbc6e2277cd8097b33e90002844364a147e74c/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs#L406

Reply:
probably easiest to simply make sure you have tests for both with and without.

Reply:
No  I don't think we need to trim directory ending separator as this method should also detect directories.
On Unix based systems if we trim just "/" it becomes "" which is invalid path, but in reality "/" simplifies to root directory of the system.

This trimming logic was put into existing implementation because of [this](https://source.dot.net/#System.Private.CoreLib/FileSystem.Exists.Unix.cs,40) reason and I don't think we care about this anymore in this API.